### PR TITLE
Move validator types from Eudico

### PR DESCRIPTION
**Pull request by dnkolegov**
This PR adds validator-related types from Eudico.

### CODE REVIEW

File: validator/gen/gen.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package main
```
Comment (adlrocha) [inline validator/gen/gen.go:1]: Do you mind moving the generation code to `gen/gen`? Here we use a single file for code generation to make it simpler.

**Review by adlrocha — APPROVED** [review]
Do you mind changing the generation file to the root gen we use in the project? Once this is fixed feel free to merge. Thanks.

File: validator/gen/gen.go (hunk 1 above)
Comment (dnkolegov) [reply validator/gen/gen.go]: Fixed

**Review by adlrocha — CHANGES_REQUESTED** [review]
We should change in this PR already https://github.com/consensus-shipyard/go-ipc-types/blob/8dc9e20f072f17cc1a51bebbddf7c0b7868a567f/subnetactor/subnet-actor.go#L32 to be of type `ValidatorSet`

File: validator/validator.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		Addr:    a,
+		NetAddr: m.String(),
+	}, nil
+}
+
+type Validator struct {
+	Addr addr.Address `json:"addr"`
+	// FIXME: Consider using a multiaddr
+	NetAddr string `json:"net_addr,omitempty"`
+	Weight  uint64 `json:"weight"`
```
Comment (adlrocha) [inline validator/validator.go:43]: Before merging let's discuss hoe this Weight is going to be used, it is not clear to me.

Comment (adlrocha) [reply validator/validator.go:43]: Already synced on this, the field can stay, I will add the logic in the actor PR. We are just missing the comment about changing the type for the `ValidatorSet` in the state and feel free to merge.

**Comment by dnkolegov** [comment]
> We should change in this PR already
> 
> https://github.com/consensus-shipyard/go-ipc-types/blob/8dc9e20f072f17cc1a51bebbddf7c0b7868a567f/subnetactor/subnet-actor.go#L32
> 
> to be of type `ValidatorSet`

Fixed

**Pull request merged**